android/game/src/main/java/com/farsitel/bazaar/game/GameHubBridge.java

```diff
@@ -0,0 +1,422 @@
+                }
+                result.status = Status.LOGIN_CAFEBAZAAR;
+                result.message = "Login to Cafebazaar before!";
+            } catch (Exception e) {
+                e.printStackTrace();
+                result.status = Status.FAILURE;
+                result.message = e.getMessage();
+                result.stackTrace = Arrays.toString(e.getStackTrace());
+            }
+            MainThread.run(() -> {
+                callback.call(result);
+                if (showPrompts) {
+                    startActionViewIntent(context, "bazaar://login", BAZAAR_PACKAGE_NAME);
+                }
+            });
+        });
+    }
+
+    public void getTournaments(Activity activity, ITournamentsCallback callback) {
+        logger.logDebug("Call getTournaments");
+
+
+        // Check one of services is connected
+        Result serviceResult = areServicesAvailable();
+        if (serviceResult.status != Status.SUCCESS) {
+            serviceResult.call(callback);
+            return;
+        }
+
+        // Check player is already logged-in
+        isLogin(activity, false, loginResult -> {
+            if (loginResult.status != Status.SUCCESS) {
+                tournamentsCallback(loginResult, callback);
+                return;
+            }
+
+            if (isBroadcastMode) {
+                gameHubBroadcast.getTournamentTimes(tournamentResult -> {
+                    tournamentsCallback(tournamentResult, callback);
+                });
+                return;
+            }
+            executorService.submit(() -> {
+                final Result result = new Result();
+                try {
+                    Bundle bundle = gameHubService.getTournamentTimes(activity.getPackageName());
+                    result.setBundle(bundle);
+                } catch (RemoteException e) {
+                    e.printStackTrace();
+                    result.status = Status.FAILURE;
+                    result.message = e.getMessage();
+                    result.stackTrace = Arrays.toString(e.getStackTrace());
+                }
+                MainThread.run(() -> {
+                    tournamentsCallback(result, callback);
+                });
+            });
+        });
+    }
+
+    void tournamentsCallback(Result result, ITournamentsCallback callback) {
+        logger.logDebug("Tournaments: " + result.toString());
+
+        if (result.status != Status.SUCCESS) {
+            callback.onFinish(result.status.getLevelCode(), result.message, result.stackTrace, null);
+            return;
+        }
+
+        long startAt = result.extras.containsKey(ParamNames.START_TIMESTAMP) ? result.extras.getLong(ParamNames.START_TIMESTAMP) : 0;
+        long endAt = result.extras.containsKey(ParamNames.END_TIMESTAMP) ? result.extras.getLong(ParamNames.END_TIMESTAMP) : 0;
+        List<Tournament> tournaments = new ArrayList<>();
+        tournaments.add(new Tournament("-1", "Tournament -1", startAt, endAt));
```

Comment:
Remove these hardcoded strings and rename Tournament to TournamentTime it has a bad understanding and there is no way to get tournaments in the real world and confuse the developer.